main.go

```diff
@@ -193,13 +194,18 @@ func buildConnStr(resource string) string {
 // Pull API Specs from configuration
 var APILoader = APIDefinitionLoader{}
 
-func getAPISpecs() []*APISpec {
-	var apiSpecs []*APISpec
+// Currently loaded specs
+var APISpecs []*APISpec
+var apiSpecSyncMutex sync.Mutex
```

Comment:
Are you certain a lock for writing only is enough? What if there's a read while you're writing with the lock grabbed?